user-access-handlers/src/main/java/no/unit/nva/handlers/ListByInstitutionHandler.java

```diff
@@ -63,9 +69,8 @@ private boolean likeUserOrEmpty(UserDto userDto, Optional<String> userName) {
                    .orElse(true);
     }
 
-    private boolean hasRoleOrEmpty(UserDto userDto, Optional<String> roleName) {
-        return roleName
-                   .map(s -> userDto.getRoles().stream().anyMatch(f -> f.getRoleName().equals(s)))
-                   .orElse(true);
+    private boolean hasOneRole(UserDto userDto, Set<String> checkedRoles) {
```

Comment:
This returns true if there are more than zero matches between roles and user roles, but is that enough to grant access?